drafts/taking-a-byte-of-the-carbon-cake.md

```diff
@@ -0,0 +1,291 @@
+
+Fortunately there are a growing number of third-party sources which help us see the ‘wood for the trees’; one such is ClimateIQ which provides emission factors for many economic sectors, of which cloud computing is one. You can query this via their data explorer, https://www.climatiq.io/data, and sign up for a free account to run API queries and filter the data.
+
+### A worked example
+
+Using Climate IQ as a source we can determine the following carbon cost rates:
+
+| Carbon Cost Item         | Carbon Rate           | Reference                                                             |
+|--------------------------|-----------------------|-----------------------------------------------------------------------|
+| CPU compute              | 0.0006314 kg/CPU-hour | Climatiq id cpu-provider_gcp-region_europe_west_                      |
+| Memory                   | 0.0001 kgCO2e/GB-hour | Climatiq id memory-provider_gcp-region_europe_west_2                  |
+| Magnetic hard disk (HDD) | 0.0000677 kgCO2e/TB-hour | Climatiq id storage-provider_gcp_cfe-region_europe_west_2-type_hdd |
+| Solid state disk (SSD)   | 0.0003049 kgCO2e/TB-hour | Climatiq id storage-provider_gcp-region_europe_west_2-type_ssd     |
+
+Rates differ by region and cloud provider, these examples are for Google Cloud, region Europe West 2
+
+A few observations based on these rates:
+
+ * The rates ('carbon cost') are incredibly small in comparison to everyday things
+ * The rate of SSD is 4-5 times that of HDD, and in Amazon this is 20 times, and we assume this reflects the cost of electricity required to maintain state, compared to magnetic hard drive; a subject we'll touch on this in the next section
+ * The standard units for memory and disk storage are respectively GB and TB, hence the carbon rate for memory is in fact 500x that of storage at the same unit
+
+We can use these rates, or similar for other providers or regions, to calculate the cost of our use of a cloud service. We simply evaluate for each of our service resources the number of CPU hours spent, memory and disk storage per hour. A worked example from a genuine project, that hosts a modest website with some video content, is as follows:
+
+The system consists of a handful of Spring micro-services with a database, a video server and some light messaging. 
+
+Micro-services and messaging runs on a Kubernetes cluster on Google Kubernetes (GKE) over 3 nodes ( spec: n1-standard-4, 15 GB memory, storage: 100 HDD + 100 SSD).  The video platform runs on a separate server on 2 nodes ( spec: n1-standard-1, 7.5 GB memory, storage: 20 HDD and 100 SDD).  The database runs on a separate server on 1 node ( spec: n1-standard-1, 3.75 GB memory,storage 10 SSD )  
+
+The calculation is shown below and gives a total of 175 KgCO2/year for a single environment.
+
```

Comment:
add the three environments here which triples the cost per year.

and also the out of scope aspect.
as a background info on embodied emissions: the cloud carbon footprint tool takes that into consideration.
an important aspect I would als mention as out of scope is the energy consumption and the embodied emissions on the users side. depending on the usage pattern, this can outweight the data-centre emissions by far.

(optionally) with the roughly 500kg per year and the outscoped factors, you can now compare your hypothetical project to a real project: https://www.eco-compute.io/files/slides/LHIND_Determine_Software_emissions.pdf - they did a profound LCA on their software and end up with roughly 900kg per year. Which confirms, that your approach is in the right order of magnitude.